core/txpool/bundlepool/bundlepool.go

```diff
@@ -370,16 +374,27 @@ func (p *BundlePool) reset(newHead *types.Header) {
 			txSet.Add(tx.Hash())
 		}
 	}
+	var wg sync.WaitGroup
 	for hash, bundle := range p.bundles {
-		if (bundle.MaxTimestamp != 0 && newHead.Time > bundle.MaxTimestamp) ||
-			(bundle.MaxBlockNumber != 0 && newHead.Number.Cmp(new(big.Int).SetUint64(bundle.MaxBlockNumber)) > 0) {
-			p.slots -= numSlots(p.bundles[hash])
-			delete(p.bundles, hash)
-		} else if txSet.Contains(bundle.Txs[0].Hash()) {
-			p.slots -= numSlots(p.bundles[hash])
-			delete(p.bundles, hash)
-		}
+		wg.Add(1)
+		go func(hash common.Hash, bundle *types.Bundle) {
+			defer wg.Done()
+			if (bundle.MaxTimestamp != 0 && newHead.Time > bundle.MaxTimestamp) ||
+				(bundle.MaxBlockNumber != 0 && newHead.Number.Cmp(new(big.Int).SetUint64(bundle.MaxBlockNumber)) > 0) {
+				p.slots -= numSlots(p.bundles[hash])
```

Comment:
Is it thread safe to do `p.slots -=` and `delete(p.bundles, hash)`?

Reply:
The `reset` function itself is thread safe since we have 
```
	p.mu.Lock()
	defer p.mu.Unlock()
```
in the beginning of the function.

However, if you use go func here, it's not.

Reply:
Removed go func, just use single thread.